File: src/main/java/myLang/expression/RelationalExpressionContext.java
package myLang.expression;

import myLang.ContextHandler;
import myLang.response.BooleanResponse;
import myLang.response.IntegerResponse;
import myLang.response.MyLangException;
import myLang.response.Response;
import myLangParser.MyLangLexer;
import myLangParser.MyLangParser;
import org.antlr.v4.runtime.tree.TerminalNodeImpl;

public class RelationalExpressionContext implements ContextHandler<MyLangParser.RelationalExpressionContext> {

    @Override
    public Response handler(MyLangParser.RelationalExpressionContext ctx) throws MyLangException {
        if(ctx.getChildCount() == 1){
            return defaultHandler(ctx);
        }
        IntegerResponse firstResponse = (IntegerResponse) defaultHandler(ctx, 0);
        IntegerResponse secondResponse = (IntegerResponse) defaultHandler(ctx, 2);
        switch (((TerminalNodeImpl) ctx.getChild(1)).getSymbol().getType()) {
            case MyLangLexer.LT:
                return new BooleanResponse(firstResponse.getResponse() < secondResponse.getResponse());
            case MyLangLexer.GT:
                return new BooleanResponse(firstResponse.getResponse() > secondResponse.getResponse());
            case MyLangLexer.LE:
                return new BooleanResponse(firstResponse.getResponse() <= secondResponse.getResponse());
            case MyLangLexer.GE:
                return new BooleanResponse(firstResponse.getResponse() >= secondResponse.getResponse());
        }

        throw new MyLangException("Неподдерживаемая операция сравнения");
    }
}

File: src/main/kotlin/myLangKotlin/expression/RelationalExpressionContext.kt
package myLangKotlin.expression

import myLangKotlin.ContextHandler
import myLangKotlin.response.BooleanResponse
import myLangKotlin.response.IntegerResponse
import myLangKotlin.response.MyLangException
import myLangKotlin.response.Response
import myLangParser.MyLangLexer
import myLangParser.MyLangParser
import org.antlr.v4.runtime.tree.TerminalNodeImpl

class RelationalExpressionContext : ContextHandler<MyLangParser.RelationalExpressionContext> {

    @Throws(MyLangException::class)
    override fun handler(ctx: MyLangParser.RelationalExpressionContext): Response<*> {
        if (ctx.childCount == 1) {
            return defaultHandler(ctx)
        }
        val firstResponse = defaultHandler(ctx, 0) as IntegerResponse
        val secondResponse = defaultHandler(ctx, 2) as IntegerResponse
        when ((ctx.getChild(1) as TerminalNodeImpl).getSymbol().type) {
            MyLangLexer.LT -> return BooleanResponse(firstResponse.response < secondResponse.response)
            MyLangLexer.GT -> return BooleanResponse(firstResponse.response > secondResponse.response)
            MyLangLexer.LE -> return BooleanResponse(firstResponse.response <= secondResponse.response)
            MyLangLexer.GE -> return BooleanResponse(firstResponse.response >= secondResponse.response)
        }

        throw MyLangException("Неподдерживаемая операция сравнения")
    }
}
